retry_test.go

```diff
@@ -341,18 +342,18 @@ func TestRetryQuerySuccessWithTimeout(t *testing.T) {
 }
 
 func TestRetryQueryFail(t *testing.T) {
-	logger.Info("Retry N times and Fail")
+	logger.Info("Retry N times until there is a timeout and Fail")
 	client := &fakeHTTPClient{
-		cnt:     4,
-		success: false,
+		statusCode: http.StatusTooManyRequests,
+		success:    false,
 	}
 	urlPtr, err := url.Parse("https://fakeaccountretryfail.snowflakecomputing.com:443/queries/v1/query-request?" + requestIDKey)
 	if err != nil {
 		t.Fatal("failed to parse the test URL")
 	}
 	_, err = newRetryHTTP(context.Background(),
 		client,
-		emptyRequest, urlPtr, make(map[string]string), 60*time.Second, defaultTimeProvider, nil).doPost().setBody([]byte{0}).execute()
+		emptyRequest, urlPtr, make(map[string]string), 30*time.Second, defaultTimeProvider, nil).doPost().setBody([]byte{0}).execute()
```

Comment:
So this test duration is at least 30s? Maybe we can decrease it even more?

Reply:
Decreased to 15s. There have to be at least 2 retries.